geps/gep-1924.md

```diff
@@ -0,0 +1,76 @@
+
+## Extended Conformance
+
+In addition to Service, there are other optional parentRef resources that, if used by implementations, MUST adhere to the spec’s prescriptions. At the time of writing (v0.70), there are two resources in extended conformance: ServiceImport (part of the mcs-api, currently in alpha) and Gateway . The semantics of ServiceImport parentRef binding can be found in GEP 1748 (Note: Headless ServiceImports are out of scope and not currently a part of the spec). The semantics for Gateways in the context of GAMMA implementations are laid out below:
+
+### `Gateways`
+
+There has been much discussion around cluster local Gateways (i.e. Gateways not associated with a traditional load balancer). While there are various potential UX impairments (e.g. what’s the difference between a GAMMA HTTPRoute with a Gateway parentRef and an ingress implementation’s HTTPRoute?), there is no technical reason why a Gateway cannot be a valid GAMMA parentRef. In this case, the mesh implementation MUST use the ClusterIP of the Gateway for traffic matching (but not necessarily to the exclusion of the External IP).
+
+### Why not `IPAddress`
+
+In Kubernetes 1.27, there will be a new IPAddress resource added to networking.k8s.io/v1alpha1 as part of [KEP 1880](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/1880-multiple-service-cidrs#proposal). Naturally, it makes sense to examine whether or not this new resource makes sense as a GAMMA aware parentRef. At first glance, IPAddress seems to be an appropriate abstraction for the “frontend” role we’ve been discussing; every Kubernetes Service is accessed over the network via one of its ip addresses. Furthermore, the fact that the Service resource auto-creates an IPAddress is encouraging. However, the fact that the name of the IPAddress is simply the decimal/hex ip address and not a human-readable Service name makes the UX untenable as a parentRef.
+
+## Implementation-specific `parentRef`s
+
+The remainder of this GEP will detail considerations and guidelines for mesh implementations if they wish to enable an implementation-specific resource as a parentRef. Recall that the frontend role of a (generic) service is how one calls the service. In the service mesh transparent proxy context, the frontend role (and parentRef by extension) is effectively the matching mechanism for the specified route. For the Service parentRef, this means that the mesh should apply a particular xRoute’s configuration if the destination ip address for a given connection is the ClusterIP of that parentRef Service. If a mesh wishes to use an implementation-specific resource for parentRef, that resource MUST contain layer-appropriate information suitable for traffic matching (e.g. no Host header capture in TCPRoute). For example, consider the following HTTPRoute with an Istio ServiceEntry as a parentRef:
+
+```yaml
+apiVersion: networking.istio.io/v1beta1
+kind: ServiceEntry
```

Comment:
I am trying to convince people to push the "hosts" portion of ServiceEntry to MCS ServiceExport - making it possible to define 'bring your own' names. This may allow us to clean up ServiceEntry, which is now 
a mix of egress, internal and deprecated fields ( WorkloadGroup/etc).

Note that your example should be location: MESH_INTERNAL - the MESH_EXTERNAL example is an egress use case which we agreed to postpone and decouple for now.

Reply:
Changed everything to internal